From 50bd3f64924a56aa11bfcaebfe1efd705032ff92 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 26 Aug 2014 13:11:38 -0400
Subject: [redhat] kernel.spec: Remove trace from packaged perf binaries

Message-id: <20140826131138.GA27000@krava.brq.redhat.com>
Patchwork-id: 89109
O-Subject: [PATCH] perf spec: Remove trace from packaged binaries packaged binaries
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
RHEL7.1 specific

description
===========
The v3.12 perf tool code adds 'trace' as another binary.
Fedora spec is not packaging it with following reason:

  Don't install the 'trace' alias symlink for 'perf trace'.  Seems much too
  generic to just throw that into /usr/bin.

Following Fedora decision until we find some agreement.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/kernel.spec.template b/redhat/kernel.spec.template
index 9a29e52..e8d1c3d 100644
--- a/redhat/kernel.spec.template
+++ b/redhat/kernel.spec.template
@@ -1172,6 +1172,8 @@ cp %{SOURCE23} %{SOURCE24} %{SOURCE25} $INSTALL_KABI_PATH
 %if %{with_perf}
 # perf tool binary and supporting scripts/binaries
 %{perf_make} DESTDIR=$RPM_BUILD_ROOT install
+# remove the 'trace' symlink.
+rm -f $RPM_BUILD_ROOT/%{_bindir}/trace
 
 # perf-python extension
 %{perf_make} DESTDIR=$RPM_BUILD_ROOT install-python_ext
-- 
1.7.1